### PR TITLE
Labryrinth - Sandra and Karan

### PR DESCRIPTION
**Pull request by karanmann**
Link to my deployed project: https://labyrinth-react-redux.netlify.app

This weeks project was made by Sandra(https://github.com/sandra-shumanteva-kaevska) and me. Our reflections on how this week's project turned out: It was a tough project and we really had to push our limits on understanding how the dispatches and actions effected in Redux Toolkit.

### CODE REVIEW

File: code/src/index.css [hunk 1]
```diff
@@ -5,9 +5,32 @@ body {
...
   -webkit-font-smoothing: antialiased;
   -moz-osx-font-smoothing: grayscale;
+  background-image: url(./assets/background.jpg);
 }
 
 code {
   font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
     monospace;
 }
+
```
Comment (mhammarback) [inline code/src/index.css:15]: delete commented code

File: code/src/reducers/games.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		username: '',
+		actions: [],
+		description: '',
+		coordinates: '',
+		history: [],
+		currentStep:{}
+	},
+	reducers: {
+
+		history: state => {
```
Comment (mhammarback) [inline code/src/reducers/games.js:17]: yay with history!

File: code/src/reducers/games.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+			store.coordinates = action.payload.coordinates
+		},
+
+		setUsername: (store, action) => {
+			store.username = action.payload
+		}
+	}
+})
+
+export const thunk = (userName) => {
```
Comment (mhammarback) [inline code/src/reducers/games.js:48]: It would be nice to have the thunks/fetches in its own file

**Review by mhammarback — CHANGES_REQUESTED** [review]
The code was extra hard to read because of the indentation, we are requesting changes as reminder to review your code before turning it in :) 

Requested changes: 
- remove commented out code 
-  review indentation ( all pages ) 
- remove extra line spaces
- It's great that you used material ui, but maybe add some extra margin for the direction buttons. 
- Your page response to different devices, however, it is not styled for different devices. For exempel, any extra padding and margin on top should be reviewed for display on mobil. 

File

We know you had some trouble getting started this week but you where able to successfully
produce the labyrinth app, including the blue level and even some of the red. Great work Karan and Sandra!

**Comment by karanmann** [comment]
Thank you Moa and Peggy for the review. 
Yes this week was a bit messed up with me being distracted. 
We have gone ahead and made the requested changes now :)

**Comment by puj** [comment]
Great job!  
I love that you used Material UI, adds so much value.  I think Moa's requested change for the buttons still needs to be adjusted
image

My change request is:
- If you are going to show the loading icon while you are displayed the actions for the current room, disable the buttons (you can still click them while the loader is visible)
- Maybe move the loader below the description/actions so the text doesn't "jump" when the loader shows/hides

**Review by puj — CHANGES_REQUESTED** [review]
See my comment with 2 changes requested

**Comment by karanmann** [comment]
> Great job!
> I love that you used Material UI, adds so much value. I think Moa's requested change for the buttons still needs to be adjusted
> image
> 
> My change request is:
> 
> * If you are going to show the loading icon while you are displayed the actions for the current room, disable the buttons (you can still click them while the loader is visible)
> * Maybe move the loader below the description/actions so the text doesn't "jump" when the loader shows/hides


Hi Van,

I have done the changes as requested. About the button. We had decided to go with button grouping and use the template accordingly. I had conveyed the same to Moa and Peggy too. If you still think we should change it we could do it.

best regards
Sandra & Karan

**Comment by puj** [comment]
The changes turned out really good! Great job! Approving!  Nice choice with the disable style!

**Review by puj — APPROVED** [review]

**Pull request closed without merging**